nt-be/src/main.rs

```diff
@@ -21,6 +22,64 @@ async fn main() {
             .expect("Failed to initialize application state"),
     );
 
+    // Spawn background monitoring task
+    let state_clone = state.clone();
+    tokio::spawn(async move {
+        use near_api::Chain;
+        use nt_be::handlers::balance_changes::account_monitor::run_monitor_cycle;
+
+        // Get monitoring interval from env or default to 5 minutes
+        let interval_minutes: u64 = std::env::var("MONITOR_INTERVAL_MINUTES")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            .unwrap_or(5);
+
+        let interval = Duration::from_secs(interval_minutes * 60);
+
+        log::info!(
+            "Starting background monitoring service (interval: {} minutes)",
+            interval_minutes
+        );
+
+        // Wait a bit before first run to let server fully start
+        tokio::time::sleep(Duration::from_secs(10)).await;
+
+        loop {
+            log::info!("Running monitoring cycle...");
+
+            // Get current block height from the network
+            let up_to_block = match Chain::block().fetch_from(&state_clone.network).await {
+                Ok(block) => block.header.height as i64,
+                Err(e) => {
+                    log::error!("Failed to get current block height: {}", e);
+                    log::info!("Retrying in {} minutes", interval_minutes);
+                    tokio::time::sleep(interval).await;
+                    continue;
+                }
+            };
+
+            log::info!("Processing up to block {}", up_to_block);
+
+            match run_monitor_cycle(
+                &state_clone.db_pool,
+                &state_clone.archival_network,
+                up_to_block,
+            )
+            .await
+            {
+                Ok(()) => {
+                    log::info!("Monitoring cycle completed successfully");
+                }
+                Err(e) => {
+                    log::error!("Monitoring cycle failed: {}", e);
+                }
+            }
+
+            log::info!("Next monitoring cycle in {} minutes", interval_minutes);
+            tokio::time::sleep(interval).await;
```

Comment:
Personally, I prefer https://docs.rs/tokio/latest/tokio/time/fn.interval.html
As it includes time for processing in those 5 minutes, you are really running once per five minutes or immediatly if it took more than 5 mins rather than guaranteeing sleep